library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/process/traversal/util/GafferPredicateFactory.java

```diff
@@ -69,13 +75,20 @@ public static Predicate<?> getKoryphePredicate(final P<?> p) {
         }
 
         BiPredicate<?, ?> biPredicate = p.getBiPredicate();
-        Object value = p.getValue();
         if (biPredicate instanceof Compare) {
+            Object value = p.getValue();
+            if (value instanceof String) {
```

Comment:
Would it be better to have this check in the utility method to reduce duplication and then just rename it to something like `parseObjectAsTstvIfValid`?